ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/sharing/AppSharingPresenter.java

```diff
@@ -0,0 +1,341 @@
+/**
+ * @author sriram
+ */
+package org.iplantc.de.apps.client.presenter.sharing;
+
+import org.iplantc.de.apps.client.views.sharing.AppSharingView;
+import org.iplantc.de.client.models.apps.App;
+import org.iplantc.de.client.models.collaborators.Collaborator;
+import org.iplantc.de.client.models.diskResources.PermissionValue;
+import org.iplantc.de.client.models.sharing.SharedResource;
+import org.iplantc.de.client.models.sharing.Sharing;
+import org.iplantc.de.client.services.AppUserServiceFacade;
+import org.iplantc.de.client.sharing.SharingPermissionsPanel;
+import org.iplantc.de.client.sharing.SharingPresenter;
+import org.iplantc.de.client.util.DiskResourceUtil;
+import org.iplantc.de.client.util.JsonUtil;
+import org.iplantc.de.collaborators.client.util.CollaboratorsUtil;
+import org.iplantc.de.commons.client.ErrorHandler;
+import org.iplantc.de.commons.client.info.IplantAnnouncer;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONString;
+import com.google.gwt.json.client.JSONValue;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HasOneWidget;
+
+import com.sencha.gxt.core.shared.FastMap;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AppSharingPresenter implements SharingPresenter {
+
+    private final class LoadPermissionsCallback implements AsyncCallback<String> {
+        private final class GetUserInfoCallback implements AsyncCallback<FastMap<Collaborator>> {
+            private final List<String> usernames;
+
+            private GetUserInfoCallback(List<String> usernames) {
+                this.usernames = usernames;
+            }
+
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(FastMap<Collaborator> results) {
```

Comment:
I think it has the potential to be confusing that a lot of the internal variables in this function reference data sharing and disk resources rather than apps; reading the source of parseNameFromPath it seems like that shouldn't be necessary here, either, but I may be misunderstanding what it does (but I'm not sure what a 'path' would constitute, for an app)